### PR TITLE
maybe_total_supply in pallet-assets

### PR DESCRIPTION
**Pull request by girazoki**
Adds _maybe_total_supply_ to to pallet-assets functions. The previous _total_supply_ function does not allow us to distinguish between an assetId that does not exist or an assetId that exists but for which no tokens were minted.

The new _maybe_total_supply_  allows us to receive, similar to _maybe_balance_, an Option value. This would return None for a non-existent assetId.

### CODE REVIEW

File: frame/assets/src/functions.rs [hunk 1]
```diff
@@ -55,7 +55,12 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 	/// Get the total supply of an asset `id`.
 	pub fn total_supply(id: T::AssetId) -> T::Balance {
-		Asset::<T, I>::get(id).map(|x| x.supply).unwrap_or_else(Zero::zero)
+		Self::maybe_total_supply(id).unwrap_or_else(Zero::zero)
```
Comment (ggwpez) [inline frame/assets/src/functions.rs:58]: Just wondering: why was it not `unwrap_or_default` before?

**Review by ggwpez — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

File: frame/assets/src/functions.rs (hunk 1 above)
Comment (bkchr) [reply frame/assets/src/functions.rs:58]: But also fine this way.

Comment (bkchr) [inline frame/assets/src/functions.rs:58]:
```suggestion
		Self::maybe_total_supply(id).unwrap_or_default()
```

**Comment by bkchr** [comment]
bot merge

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**